File: sdk/approute-sdk/python/approute/models/accounts.py
from datetime import datetime

from approute.models._base import _Base


class AccountActivity(_Base):
    id: str
    currency: str
    amount: float
    operation: str
    created_at: datetime


class Account(_Base):
    currency: str
    balance: float
    available: float
    overdraft_limit: float
    recent_activity: list[AccountActivity] = []


class AccountListResponse(_Base):
    items: list[Account]


class AccountTransaction(_Base):
    id: str
    currency: str
    transaction_id: str
    category: str
    balance: float
    amount: float
    order_id: str
    order_id_raw: str | None = None
    description: str | None = None
    created_at: datetime


class AccountTransactionPage(_Base):
    total_count: int
    items: list[AccountTransaction]

File: sdk/approute-sdk/golang/model/accounts.go
package model

import (
	"time"

	"github.com/approute/public-api-sdk-go/enum"
)

// AccountActivity represents a single recent balance movement.
type AccountActivity struct {
	ID        string    `json:"id"`
	Currency  string    `json:"currency"`
	Amount    float64   `json:"amount"`
	Operation string    `json:"operation"`
	CreatedAt time.Time `json:"createdAt"`
}

// Account holds the balance information for a single currency.
type Account struct {
	Currency       string            `json:"currency"`
	Balance        float64           `json:"balance"`
	Available      float64           `json:"available"`
	OverdraftLimit float64           `json:"overdraftLimit"`
	RecentActivity []AccountActivity `json:"recentActivity"`
}

// AccountListResponse is the top-level response for GET /accounts.
type AccountListResponse struct {
	Items []Account `json:"items"`
}

// AccountTransaction represents one balance-transaction record.
type AccountTransaction struct {
	ID            string               `json:"id"`
	Currency      string               `json:"currency"`
	TransactionID string               `json:"transactionId"`
	Category      enum.BalanceCategory `json:"category"`
	Balance       float64              `json:"balance"`
	Amount        float64              `json:"amount"`
	OrderID       string               `json:"orderId"`
	OrderIDRaw    *string              `json:"orderIdRaw,omitempty"`
	Description   *string              `json:"description,omitempty"`
	CreatedAt     time.Time            `json:"createdAt"`
}

// AccountTransactionPage is the paginated response for GET /accounts/transactions.
type AccountTransactionPage struct {
	TotalCount int                  `json:"totalCount"`
	Items      []AccountTransaction `json:"items"`
}
